docs/workflow/building/coreclr/README.md

```diff
@@ -1,100 +1,156 @@
+
+Note that this package only includes the runtime, therefore you will only be able to run apps but not build them. For that, you would need the full SDK.
+
+**NOTE:** On Windows, this will also include `.exe` and `.msi` installers, which you can use in case you want to test your build machine-wide. This is the closest you can get to an official build installation.
+
+For a full guide on using the shipping packages for testing checkout the doc we have dedicated to it [over here](/docs/workflow/testing/using-dev-shipping-packages.md).
 
-We recommend using _Ninja_ for building the project on Windows since it more efficiently uses the build machine's resources for the native runtime build in comparison to Visual Studio's _MSBuild_.
+### Cross Compilation
+
+Using an x64 machine, it is possible to generate builds for other architectures. Not all architectures are supported for cross-compilation however, and it's also dependent on the OS you are using to build and target. Refer to the table below for the compatibility matrix.
+
+| Operating System | To x86   | To Arm32 | To Arm64 |
+| :--------------: | :------: | :------: | :------: |
+| Windows          | &#x2714; |          | &#x2714; |
+| macOS            |          |          | &#x2714; |
+| Linux            |          | &#x2714; | &#x2714; |
+
+**NOTE:** On macOS, it is also possible to cross-compile from ARM64 to x64 using an Apple Silicon Mac.
+
+Detailed instructions on how to do cross-compilation can be found in the cross-building doc [over here](/docs/workflow/building/coreclr/cross-building.md).
+
+## Other Features
+
+### Build Drivers
+
+By default, the CoreCLR build uses *Ninja* as the native build driver on Windows, and *Make* on non-Windows platforms. You can override this behavior by passing the appropriate flags to the build script:
+
+To use Visual Studio's *MSBuild* instead of *Ninja* on Windows:
+
+```cmd
+./build.cmd -subset clr -msbuild
+```
+
+It is recommended to use *Ninja* on Windows, as it uses the build machine's resources more efficiently in comparison to Visual Studio's *MSBuild*.
+
+To use *Ninja* instead of *Make* on non-Windows:
+
+```bash
+./build.sh -subset clr -ninja
+```
 
 ### Extra Flags
 
-To pass extra compiler/linker flags to the coreclr build, set the environment variables `EXTRA_CFLAGS`, `EXTRA_CXXFLAGS` and `EXTRA_LDFLAGS` as needed. Don't set `CFLAGS`/`CXXFLAGS`/`LDFLAGS` directly as that might lead to configure-time tests failing.
+You can also pass some extra compiler/linker flags to the CoreCLR build. Set the `EXTRA_CFLAGS`, `EXTRA_CXXFLAGS`, and `EXTRA_LDFLAGS` as you see fit for this purpose. The build script will consume them and then set the environment variables that will ultimately affect your build (i.e. those same ones without the `EXTRA_` prefix). Don't set the final ones directly yourself, as that is known to lead to potential failures in configure-time tests.
```

Comment:
Instead of just mentioning these, do we want to give a scenario in which it would be beneficial to use the extra flags?

Reply:
Will address this in a follow-up PR.